src/digestif_conv.ml

```diff
@@ -68,6 +73,11 @@ module Make (D : sig val digest_size : int end) = struct
     if !offset + D.digest_size = String.length str
     then res else invalid_arg "Too much enough bytes (reach: %d, expect: %d)" (!offset + (D.digest_size * 2)) (String.length str)
 
+  let consistent_of_hex_opt hex =
+    match consistent_of_hex hex with
+    | digest -> Some digest
+    | exception _ -> None
```

Comment:
We should probably match on `Invalid_arg` as catch-all clauses can be a bit dangerous (and catch valid errors like Sys.break)

Reply:
Very good point - sorry for the over-general catch-all, and thank you @dinosaure for fixing it in #66